docs: add notice of early response in hooks (#3135) (#3138)
fastify/fastify: docs/Routes.md

  `reply` is defined in [Reply](Reply.md).

> **Notice:** The documentation of `onRequest`, `preParsing`, `preValidation`, `preHandler`, `preSerialization`, `onSend`, and `onResponse` are described in more details in [Hooks](Hooks.md) and to send response before request handled by `handler`, you need to refer [Respond to a request from a hook](Hooks.md#respond-to-a-request-from-a-hook) section for best practices.

Reviewer: ```suggestion
**Notice:** The documentation of `onRequest`, `preParsing`, `preValidation`,
`preHandler`, `preSerialization`, `onSend`, and `onResponse` are described in
more detail in [Hooks](Hooks.md). Additionally, to send a response before the
request is handled by the `handler` please refer to
[Respond to a request from a hook](Hooks.md#respond-to-a-request-from-a-hook).
```
